### PR TITLE
tickv: Remove the offset from read_region()

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

read_region() originally had a `offset` argument. The idea was to avoid
reading an entire flash region and only read part of the region, by
offsetting into the region.

This was never actually supported, the idea was to be able to support it
without API breakages, but in practice I don't think that will be
possible as we will need some API changes no matter what.

As offset is never used let's just remove it. This will simplify things
a little bit and it's easy to re-add in the future if we want to support
partial page reads.

This also fixes https://github.com/tock/tock/issues/4066

### Testing Strategy

CI

### TODO or Help Wanted

N/A

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

**Review by lschuermann — DISMISSED** [review]

File: libraries/tickv/src/crc32.rs [hunk 1]
```diff
@@ -352,6 +352,12 @@ impl Crc {
     }
 }
 
+impl Default for Crc32 {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
```
Comment (bradjc) [inline libraries/tickv/src/crc32.rs:360]: ?

**Review by bradjc — COMMENTED** [review]
Looks good other than the crc32 change.

File: libraries/tickv/src/crc32.rs (hunk 1 above)
Comment (alistair23) [reply libraries/tickv/src/crc32.rs:360]: It's a clippy lint

**Review by ppannuto — DISMISSED** [review]
Weird that clippy didn't bug about this till now, but it's in its own commit which looks fine 🤷🏼

**Comment by bradjc** [comment]
That isn't a clippy lint we use though.

**Comment by alistair23** [comment]
If you run `cargo clippy` in the directory it warns about it. It's adding a single `Default` as [recommended](https://rust-lang.github.io/rust-clippy/v0.0.212/#new_without_default) by the Rust ecosystem. Why is this controversial?

**Comment by bradjc** [comment]
I would like to fix that (#4082).

As a PR reviewer it is a lot harder when there are unrelated changes, as I expect to review tickv read code but also have to ask why not just the `Default` derive? Also PRs that have unrelated changes lead to confusing history.

**Comment by lschuermann** [comment]
Marking last-call. I think regardless of the outcome of the Clippy discussion, it's probably fine to have Default be implemented for Crc32, so the changes themselves here seem uncontroversial.

**Review by bradjc — CHANGES_REQUESTED** [review]
Why not #derive default?

**Comment by alistair23** [comment]
> Why not #derive default?

Clippy recommended this implementation, so I went with that. I have swapped it to `#[derive(Default)]`

**Review by lschuermann — APPROVED** [review]

**Comment by alevy** [comment]
@bradjc I think your comment is resolved?

**Review by alevy — APPROVED** [review]

**Review by bradjc — APPROVED** [review]

**Pull request merged**